### PR TITLE
kernel: bump 5.15 to 5.15.104

### PR DESCRIPTION
**Pull request by graysky2**
All patches automatically rebased.

Build system: x86_64
Build-tested: bcm2711/RPi4B, ramips/tplink_archer-a6-v3, filogic/xiaomi_redmi-router-ax6000-ubootmod
Run-tested: bcm2711/RPi4B, ramips/tplink_archer-a6-v3, filogic/xiaomi_redmi-router-ax6000-ubootmod

### CODE REVIEW

**Comment by graysky2** [comment]
As-is, building fails for:
```
CONFIG_TARGET_ramips=y
CONFIG_TARGET_ramips_mt7621=y
CONFIG_TARGET_ramips_mt7621_DEVICE_tplink_archer-a6-v3=y
```

Here is the failure:
```
...
make[6]: Entering directory '/scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103'
make[7]: Entering directory '/scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103'
make[8]: Entering directory '/scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103'
make[9]: Entering directory '/scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103'
  CC      drivers/net/ethernet/mediatek/mtk_eth_soc.o
drivers/net/ethernet/mediatek/mtk_eth_soc.c: In function 'mtk_mac_config':
drivers/net/ethernet/mediatek/mtk_eth_soc.c:503:9: error: 'mcr_cur' undeclared (first use in this function)
  503 |         mcr_cur = mtk_r32(mac->hw, MTK_MAC_MCR(mac->id));
      |         ^~~~~~~
drivers/net/ethernet/mediatek/mtk_eth_soc.c:503:9: note: each undeclared identifier is reported only once for each function it appears in
drivers/net/ethernet/mediatek/mtk_eth_soc.c:504:9: error: 'mcr_new' undeclared (first use in this function)
  504 |         mcr_new = mcr_cur;
      |         ^~~~~~~
drivers/net/ethernet/mediatek/mtk_eth_soc.c: In function 'mtk_mac_finish':
drivers/net/ethernet/mediatek/mtk_eth_soc.c:531:22: warning: unused variable 'mcr_new' [-Wunused-variable]
  531 |         u32 mcr_cur, mcr_new;
      |                      ^~~~~~~
drivers/net/ethernet/mediatek/mtk_eth_soc.c:531:13: warning: unused variable 'mcr_cur' [-Wunused-variable]
  531 |         u32 mcr_cur, mcr_new;
      |             ^~~~~~~
make[9]: *** [scripts/Makefile.build:289: drivers/net/ethernet/mediatek/mtk_eth_soc.o] Error 1
make[9]: Leaving directory '/scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103'
make[8]: *** [scripts/Makefile.build:552: drivers/net/ethernet/mediatek] Error 2
make[8]: Leaving directory '/scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103'
make[7]: *** [scripts/Makefile.build:552: drivers/net/ethernet] Error 2
make[7]: Leaving directory '/scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103'
make[6]: *** [scripts/Makefile.build:552: drivers/net] Error 2
make[6]: Leaving directory '/scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103'
make[5]: *** [Makefile:1912: drivers] Error 2
make[5]: Leaving directory '/scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103'
make[4]: *** [Makefile:23: /scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103/.modules] Error 2
make[4]: Leaving directory '/scratch/union/target/linux/ramips'
make[3]: *** [Makefile:11: compile] Error 2
make[3]: Leaving directory '/scratch/union/target/linux'
time: target/linux/compile#2.52#0.71#3.13
    ERROR: target/linux failed to build.
make[2]: *** [target/Makefile:30: target/linux/compile] Error 1
make[2]: Leaving directory '/scratch/union'
make[1]: *** [target/Makefile:23: /scratch/union/staging_dir/target-mipsel_24kc_musl/stamp/.target_compile] Error 2
make[1]: Leaving directory '/scratch/union'
make: *** [/scratch/union/include/toplevel.mk:231: world] Error 2
```

Can someone take a look/propose change to fix?  @hauke @robimarko ?

File: target/linux/generic/backport-5.15/704-15-v5.19-net-mtk_eth_soc-move-MAC_MCR-setting-to-mac_finish.patch [hunk 1]
```diff
@@ -25,15 +25,16 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
...
  		return;
  	}
  
 -	/* Setup gmac */
 -	mcr_cur = mtk_r32(mac->hw, MTK_MAC_MCR(mac->id));
 -	mcr_new = mcr_cur;
 -	mcr_new |= MAC_MCR_IPG_CFG | MAC_MCR_FORCE_MODE |
--		   MAC_MCR_BACKOFF_EN | MAC_MCR_BACKPR_EN | MAC_MCR_FORCE_LINK;
+-		   MAC_MCR_BACKOFF_EN | MAC_MCR_BACKPR_EN | MAC_MCR_FORCE_LINK |
+-		   MAC_MCR_RX_FIFO_CLR_DIS;
```
Comment (hauke) [inline target/linux/generic/backport-5.15/704-15-v5.19-net-mtk_eth_soc-move-MAC_MCR-setting-to-mac_finish.patch:37]: You have to add `MAC_MCR_RX_FIFO_CLR_DIS` to the block where it was moved to about 20 lines below this line.

Comment (graysky2) [reply target/linux/generic/backport-5.15/704-15-v5.19-net-mtk_eth_soc-move-MAC_MCR-setting-to-mac_finish.patch:37]: I not sure exactly what you're asking me to add... would you mind sending a mergable snippet here as you did last time?

Comment (hauke) [reply target/linux/generic/backport-5.15/704-15-v5.19-net-mtk_eth_soc-move-MAC_MCR-setting-to-mac_finish.patch:37]: This is missing:
````
--- a/target/linux/generic/backport-5.15/704-15-v5.19-net-mtk_eth_soc-move-MAC_MCR-setting-to-mac_finish.patch
+++ b/target/linux/generic/backport-5.15/704-15-v5.19-net-mtk_eth_soc-move-MAC_MCR-setting-to-mac_finish.patch
@@ -43,7 +43,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
  	return;
  
  err_phy:
-@@ -478,6 +467,26 @@ init_err:
+@@ -478,6 +467,27 @@ init_err:
  		mac->id, phy_modes(state->interface), err);
  }
  
@@ -58,7 +58,8 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
 +	mcr_cur = mtk_r32(mac->hw, MTK_MAC_MCR(mac->id));
 +	mcr_new = mcr_cur;
 +	mcr_new |= MAC_MCR_IPG_CFG | MAC_MCR_FORCE_MODE |
-+		   MAC_MCR_BACKOFF_EN | MAC_MCR_BACKPR_EN | MAC_MCR_FORCE_LINK;
++		   MAC_MCR_BACKOFF_EN | MAC_MCR_BACKPR_EN | MAC_MCR_FORCE_LINK |
++		   MAC_MCR_RX_FIFO_CLR_DIS;
 +
 +	/* Only update control register when needed! */
 +	if (mcr_new != mcr_cur)
````

Comment (graysky2) [reply target/linux/generic/backport-5.15/704-15-v5.19-net-mtk_eth_soc-move-MAC_MCR-setting-to-mac_finish.patch:37]: Thank you for the patch, applied.

**Comment by graysky2** [comment]
I repeated the build-test/run-test for all three in my original commit message.  All are good.

**Comment by graysky2** [comment]
Some recent commits must have caused this breakage (27 failing and 198 successful).  I am re-running the update and will force push.  Should take care of it.

**Pull request merged**